src/main/c/NativeStatus.cpp

```diff
@@ -0,0 +1,38 @@
+#include "hail/NativeStatus.h"
+#include "hail/NativePtr.h"
+#include <jni.h>
+
+namespace hail {
+
+NATIVEMETHOD(long, NativeStatus, nativeCtorErrnoOffset)(
+  JNIEnv* env,
+  jobject thisJ
+) {
+  auto status = std::make_shared<NativeStatus>();
+  // C++ "offsetof" can be weird when used on subclasses
```

Comment:
I think a comment link this that doesn't actually describe the issue is just going to be confusing to someone who doesn't know what the issue is.  Can you clarify or add a link to a relevant discussion?